vms/platformvm/state/disk_staker_diff_iterator.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	// startDiffKey = [subnetID] + [inverseHeight]
+	startDiffKeyLength = ids.IDLen + database.Uint64Size
+	// diffKey = [subnetID] + [inverseHeight] + [nodeID]
+	diffKeyLength       = startDiffKeyLength + ids.NodeIDLen
+	diffKeyHeightOffset = ids.IDLen
+	diffKeyNodeIDOffset = diffKeyHeightOffset + database.Uint64Size
+
+	// weightValue = [isNegative] + [weight]
+	weightValueLength = database.BoolSize + database.Uint64Size
+)
+
+var (
+	errUnexpectedDiffKeyLength     = fmt.Errorf("expected diff key length %d", diffKeyLength)
+	errUnexpectedWeightValueLength = fmt.Errorf("expected weight value length %d", weightValueLength)
+)
+
+// getStartDiffKey is used to determine the starting key when iterating.
+//
+// Note: the result should be a prefix of [getDiffKey] if called with the same
```

Comment:
nit: can "should" be "is"? I assume we're stating an invariant

Reply:
updated the `note` to an `invariant`